fl4health/preprocessing/autoencoders/dim_reduction.py

```diff
@@ -0,0 +1,140 @@
+from abc import ABC
+from pathlib import Path
+
+import torch
+
+
+class AutoEncoderProcessing(ABC):
+    def __init__(
+        self,
+        checkpointing_path: Path,
+    ) -> None:
+        """
+        Abstract class for processors that work with a pre-trained and saved autoencoder model.
+
+        Args:
+            checkpointing_path (Path): Path to the saved model.
+        """
+        self.checkpointing_path = checkpointing_path
+        self.device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
```

Comment:
I think we actually want to have the user pass this value into the class (we can set a default if it isn't passed. However, consider the case where there is a GPU available, but the user doesn't want to use it. So everywhere else they set device to CPU, but we'll get a mismatch here.